solvers/chop.py

```diff
@@ -0,0 +1,71 @@
+import warnings
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import torch
+    import chop
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
```

Comment:
This call should be put in `set_objective`

As it is now, it would prevent `benchopt` from importing the class in an emv where pytorch is not installed.
 This is for instance the case when collecting the requirements to install them.

Reply:
actually, device is not used?

Reply:
Now it is!